src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,78 @@
-export const Pagination = () => {};
+import React from 'react';
+import { getNumbers } from '../../utils';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (val: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
```

Comment:
Consider memoizing the `pages` array using `useMemo` to avoid recalculating it on every render when the `total` and `perPage` values haven't changed.